### PR TITLE
Add ability to manually disconnect for Sftp driver

### PR DESCRIPTION
**Pull request by hasfoug**
Exposed ConnectionProvider in SftpAdapter via geter, added disconnect method in SftpConnectionProvider

Related issues: 
https://github.com/thephpleague/flysystem/issues/1660
https://github.com/thephpleague/flysystem/issues/1757

### CODE REVIEW

File: src/PhpseclibV3/SftpAdapter.php [hunk 1]
```diff
@@ -47,6 +47,11 @@ public function __construct(
         $this->mimeTypeDetector = $mimeTypeDetector ?? new FinfoMimeTypeDetector();
     }
 
+    public function getConnectionProvider(): ConnectionProvider
```
Comment (frankdejonge) [inline src/PhpseclibV3/SftpAdapter.php:50]: Flysystem is not a DI, this should be removed.

Comment (hasfoug) [reply src/PhpseclibV3/SftpAdapter.php:50]: @frankdejonge But how then i should call disconnect from SftpAdapter if ConnectionProvider is private? Do you suggest that we should rather add a public disconnect method in SftpAdapter as well?

Comment (frankdejonge) [reply src/PhpseclibV3/SftpAdapter.php:50]: @hasfoug the connection provider is injected in, so you can get the same instance somewhere else.

Comment (ttruong15) [reply src/PhpseclibV3/SftpAdapter.php:50]: @frankdejonge unfortunately the connection provider is injected in but for Laravel the way it's call the connection provider is a local variable and there's no way to access it.

Here's where it's being called:

vendor/laravel/framework/src/Illuminate/Filesystem/FilesystemManager.php

`    /**
     * Create an instance of the sftp driver.
     *
     * @param  array  $config
     * @return \Illuminate\Contracts\Filesystem\Filesystem
     */
    public function createSftpDriver(array $config)
    {
        $provider = SftpConnectionProvider::fromArray($config);

        $root = $config['root'] ?? '/';

        $visibility = PortableVisibilityConverter::fromArray(
            $config['permissions'] ?? []
        );

        $adapter = new SftpAdapter($provider, $root, $visibility);

        return new FilesystemAdapter($this->createFlysystem($adapter, $config), $adapter, $config);
    }`
    
    So here $provider is a local variable so we have no way off accessing the provider.  So if i want to expose the provider in Laravel I have to implement my own Storage system just to override this method to expose the provider. Such a pain.

**Comment by frankdejonge** [comment]
Superseded by #1763

**Pull request closed without merging**